MUI `Link` styling regression with updated `@emotion/` and `@mui/` packages

## Context:
`@mui/lab` was added as a dependency and versions of `@emotion/react` `@emotion/styled` and `@mui/material` were bumped in order to use the MUI Lab [Tabs API](https://mui.com/material-ui/react-tabs/#experimental-api) when implementing `MetricChartTabs` (PR #479). 


## Issue: 

Presumably, updating package versions broke/changed the use of MUI `createStyled()` on the MUI `Link` component. No other components were affected: 

    .../MetricCompareTable.style.ts(12,14): error TS2742: The inferred type of 'StyledLink' cannot be named without a reference to '@mui/system/node_modules/csstype'. This is likely not portable. A type annotation is necessary.

VSCode is still able to infer the type:
![](https://user-images.githubusercontent.com/55333380/207084258-3659bc92-b681-4d9a-85d3-50fa8e5d140d.png)

but the build failed. Per the error, adding an explicit (though incomplete) type annotation fixed the build, but this doesn’t feel like a satisfactory solution and doesn’t get at the root cause of the issue. Removing the styled component and adding styles inline also circumvented the issue. 


## Package Updates:
| package         | old version | new version      |
| --------------- | ----------- | ---------------- |
| @emotion/react  | ^11.9.3     | ^11.10.5         |
| @emotion/styled | ^11.9.3     | ^11.10.5         |
| @mui/material   | ^5.9.1      | ^5.10.17         |
| @mui/lab        | --          | ^5.0.0-alpha.111 |

--- a/.changeset/early-suits-pump.md
+++ b/.changeset/early-suits-pump.md
@@ -1,0 +1,5 @@
+---
+"@actnowcoalition/ui-components": patch
+---
+
+Remove no-longer-needed type annotation on StyledLink.

--- a/packages/ui-components/src/components/MetricCompareTable/MetricCompareTable.style.ts
+++ b/packages/ui-components/src/components/MetricCompareTable/MetricCompareTable.style.ts
@@ -1,5 +1,4 @@
-import { StyledComponent } from "@emotion/styled";
-import { LinkBaseProps, Link as MuiLink } from "@mui/material";
+import { Link as MuiLink } from "@mui/material";
 
 import { styled } from "../../styles";
 import { TableCell } from "../CompareTable";
@@ -10,11 +9,7 @@ export const StyledTableCell = styled(TableCell)`
   padding: 0;
 `;
 
-// TODO: Issue with upgraded MUI/emotion requires an explicit type annotation
-// here. See issue #486
-export const StyledLink: StyledComponent<
-  Omit<LinkBaseProps, "classes">
-> = styled(MuiLink)`
+export const StyledLink = styled(MuiLink)`
   display: flex;
   padding: ${({ theme }) => theme.spacing(2)};
   text-decoration: none;

